cryptoservice/crypto_service.go

```diff
@@ -24,16 +25,16 @@ const (
 type CryptoService struct {
 	gun        string
 	passphrase string
-	keyStore   *trustmanager.KeyFileStore
+	keyStore   trustmanager.KeyStore
 }
 
 // NewCryptoService returns an instance of CryptoService
-func NewCryptoService(gun string, keyStore *trustmanager.KeyFileStore, passphrase string) *CryptoService {
+func NewCryptoService(gun string, keyStore trustmanager.KeyStore, passphrase string) *CryptoService {
 	return &CryptoService{gun: gun, keyStore: keyStore, passphrase: passphrase}
 }
 
 // Create is used to generate keys for targets, snapshots and timestamps
-func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (*data.PublicKey, error) {
+func (ccs *CryptoService) Create(role string, algorithm data.KeyAlgorithm) (data.Key, error) {
```

Comment:
Got it. A small loss for unification. I still liked the PublicKey here better. It's easier to understand what is going on.

Now just looking at the method return args it isn't obvious to the caller what is expected to get out of this function. Do I get the private key or the full public key?!